scio-core/src/main/java/com/spotify/scio/transforms/BaseAsyncDoFn.java

```diff
@@ -52,13 +52,19 @@ protected abstract FutureT addCallback(FutureT future,
 
   @StartBundle
   public void startBundle() {
+    waitForFutures();
```

Comment:
On second thought I'm not sure this would matter. Not waiting for a pending future will not cause deadlock whereas waiting for one that never completes will.